test/python/circuit/test_parameters.py

```diff
@@ -651,6 +713,46 @@ def test_num_parameters(self):
             qc.x(0)
             self.assertEqual(qc.num_parameters, 0)
 
+    def test_to_instruction_after_inverse(self):
+        """Verify converting an inverinverse generates valid ParameterTable"""
+        # ref: https://github.com/Qiskit/qiskit-terra/issues/4235
+        qc = QuantumCircuit(1)
+        theta = Parameter('theta')
+        qc.rz(theta, 0)
+
+        inv_instr = qc.inverse().to_instruction()
+        self.assertIsInstance(inv_instr, Instruction)
+
+    def test_copy_after_inverse(self):
+        """Verify circuit.inverse generates a valid ParameterTable."""
+        qc = QuantumCircuit(1)
+        theta = Parameter('theta')
+        qc.rz(theta, 0)
+
+        inv_copy = qc.inverse()
+        self.assertIn(theta, inv_copy.parameters)
+        raise_if_parameter_table_invalid(inv_copy)
+
+    def test_copy_after_mirror(self):
+        """Verify circuit.mirror generates a valid ParameterTable."""
+        qc = QuantumCircuit(1)
+        theta = Parameter('theta')
+        qc.rz(theta, 0)
+
+        mirror_copy = qc.mirror().copy()
```

Comment:
Not for this PR, but isn't it strange that `inverse` returns a copy while `mirror` doesn't? Or was the copy here not intended?

Reply:
Good catch, this was left over. The copy's not needed.

Reply:
Removed in 0952c94 .